libs/cases/frontend/view/src/lib/components/create-case/create-case.component.ts

```diff
@@ -112,18 +115,18 @@ export class CreateCaseComponent {
      * @returns {JSON}
      */
     private mapFormData(formData: CaseFormData) {
-
         return {
-            handle: 'AA-000',
+            handle: 'AA-' + faker.number.int({ min: 1000, max: 9999 }),
             dueDate: formData.dueDate,
             title: formData.title,
             type: formData.selectType,
             status: ECaseStatus.OPEN,
             description: formData.text,
-            source: 'Internal System A',
+            source: 'Internal System ' + faker.number.int({min: 1, max: 10}),
             priority: formData.selectPriority,
             createdBy: formData.email,
-            eTag: 'etag_value_here'
+            eTag: faker.string.alphanumeric(10),
         };
     }
+
 }
```

Comment:
Ich habe hier falsch gerebased die Änderungen sind von @PatrickSchm1dt. Kannst du mir sagen wie bei dir die Tests durchgelaufen sind?
Die tests sind falsch und ich weiß nicht wie du es "gefixed" hast

```ts
  ● CreateCaseComponent › should create

    NullInjectorError: R3InjectorError(Standalone[CreateCaseComponent])[XdBrowseFacade -> XdBrowseFacade -> FacilitiesRequestService -> _HttpClient -> _HttpClient]: 
      NullInjectorError: No provider for _HttpClient!

      11 | })
      12 | export class FacilitiesRequestService {
    > 13 |     private readonly _httpClient = inject(HttpClient);
         |                                          ^
      14 |     private readonly _baseRoute = '/api/facilities';
      15 |
      16 |     public getAllFacilities(): Observable<IFacilitiesResponse[]> {

      at NullInjector.get (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:1654:27)
      at R3Injector.get (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:3093:33)
      at R3Injector.get (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:3093:33)
      at injectInjectorOnly (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:1100:40)
      at ɵɵinject (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:1106:42)
      at inject (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:1192:12)
      at new FacilitiesRequestService (../../../facilities/frontend/domain/src/lib/infrastructure/facilities-request.service.ts:13:42)
      at Object.FacilitiesRequestService_Factory [as factory] (../../../../ng:/FacilitiesRequestService/ɵfac.js:5:10)
      at ../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:3219:47
      at runInInjectorProfilerContext (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:866:9)
      at R3Injector.hydrate (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:3218:21)
      at R3Injector.get (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:3082:33)
      at injectInjectorOnly (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:1100:40)
      at ɵɵinject (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:1106:42)
      at inject (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:1192:12)
      at new XdBrowseFacade (../../../facilities/frontend/domain/src/lib/application/facades/browse.facade.ts:12:40)
      at Object.XdBrowseFacade_Factory [as factory] (../../../../ng:/XdBrowseFacade/ɵfac.js:5:10)
      at ../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:3219:47
      at runInInjectorProfilerContext (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:866:9)
      at R3Injector.hydrate (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:3218:21)
      at R3Injector.get (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:3082:33)
      at R3Injector.get (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:3093:33)
      at ChainedInjector.get (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:15723:36)
      at lookupTokenUsingModuleInjector (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:5730:39)
      at getOrCreateInjectable (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:5778:12)
      at ɵɵdirectiveInject (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:11050:19)
      at ɵɵinject (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:1106:42)
      at inject (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:1192:12)
      at new CreateCaseComponent (src/lib/components/create-case/create-case.component.ts:32:44)
      at NodeInjectorFactory.CreateCaseComponent_Factory [as factory] (../../../../ng:/CreateCaseComponent/ɵfac.js:5:10)
      at getNodeInjectable (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:5984:44)
      at createRootComponent (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:15998:35)
      at ComponentFactory.create (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:15858:29)
      at initComponent (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/testing.mjs:1979:51)
      at _ZoneDelegate.invoke (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:412:32)
      at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone-testing.umd.js:2164:43)
      at _ZoneDelegate.invoke (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:411:38)
      at Object.onInvoke (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:14882:33)
      at _ZoneDelegate.invoke (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:411:38)
      at ZoneImpl.run (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:147:47)
      at NgZone.run (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/core.mjs:14733:28)
      at _TestBedImpl.createComponent (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/testing.mjs:1990:41)
      at Function.createComponent (../../../../node_modules/.pnpm/@angular+core@17.3.8_rxjs@7.8.1_zone.js@0.14.5/node_modules/@angular/core/fesm2022/testing.mjs:1783:37)
      at src/lib/components/create-case/create-case.component.spec.ts:14:21
      at fulfilled (../../../../node_modules/.pnpm/tslib@2.6.2/node_modules/tslib/tslib.js:166:62)
      at _ZoneDelegate.invoke (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:412:32)
      at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone-testing.umd.js:2164:43)
      at _ZoneDelegate.invoke (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:411:38)
      at ZoneImpl.run (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:147:47)
      at ../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:2497:42
      at _ZoneDelegate.invokeTask (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:447:37)
      at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvokeTask (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone-testing.umd.js:2195:43)
      at _ZoneDelegate.invokeTask (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:446:42)
      at ZoneImpl.runTask (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:195:51)
      at drainMicroTaskQueue (../../../../node_modules/.pnpm/zone.js@0.14.5/node_modules/zone.js/bundles/zone.umd.js:636:39)

Test Suites: 4 failed, 4 total
Tests:       4 failed, 4 total
Snapshots:   0 total
Time:        3.626 s, estimated 14 s
Ran all test suites.

————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 NX   Ran target test for project cases-frontend-view (10s)

   ✖  1/1 failed
   ✔  0/1 succeeded [0 read from cache]


 NX   Nx Cloud Warning: VCS Integration Not Configured

An active version control integration will be required to continue using Nx Cloud soon.
An organization admin can connect your organization's VCS platform.
Connect here: https://cloud.nx.app/orgs/662bfaa78b6ced3556a282a5/workspaces/662bfa1b5c61077363a02f44/settings/vcs

View structured, searchable error logs at https://cloud.nx.app/runs/JuTLQDs51k

davidschmidt@virenschleuder-

```

Reply:
Gute Frage, die Änderung hatten Patrick, Max und ich zusammen gemacht, weil vorher alle Sachen denselben Namen hatten. Die Tests haben wir gar nicht angeschaut weil sie ja automatisch bei der PR gemacht werden

Reply:
So wies aussieht, brauchen wir hier im Test noch ein provideHttpClientTesting()

Reply:
Ja das schon das probolem ist ihr erzeugt in der Datei ein new Object von einem interface welches er zu laufzeit nicht findet. Weißt du wie man das anderes als mit dem `new`machen kann oder kannst du es mal auschecken und schauen ob du die tests zum laufen bekommst?
@IngoSternberg

Reply:
Ja, ich schaue es mir heute Abend noch an

Reply:
Also auf develop habe ich zumindest den Test zum Laufen gebracht, indem ich bei den imports noch `HttpClientTestingModule` (provideHttpClientTesting() ging irgendwie nicht) hinzugefügt habe und bei den cases in die Datei `test-setup.ts` noch folgendes über das import geschrieben habe:
```
global.ResizeObserver = class ResizeObserver {
    observe() {
        // do nothing
    }
    unobserve() {
        // do nothing
    }
    disconnect() {
        // do nothing
    }
};
```

Ich weiß nicht so ganz was du mit dem `new` meinst und auch nicht warum du die Methode `mapFormData` zeigst, weil die Änderungen dort sollten doch eigentlich egal sein, oder?

Leider failen aber auch in develop noch ganz viele andere Tests. Hätte man irgendwo immer noch angeben müssen, wenn man Dateien hinzufügt, damit die Tests dazu auch von GitHub gemacht werden?

Reply:
Danke für den Fix